test/DappRegistry.ts

```diff
@@ -0,0 +1,91 @@
+import { expect } from 'chai'
+import { ethers } from 'hardhat'
+import { DappRegistry, ENSRegistry, FDSRegistrar } from '../typechain'
+
+const ZERO_HASH = '0x0000000000000000000000000000000000000000000000000000000000000000'
+
+function keccak256FromUtf8Bytes(value: string) {
+  return ethers.utils.keccak256(ethers.utils.toUtf8Bytes(value))
+}
+
+describe('dapp registry', () => {
+  let dappRegistry: DappRegistry
+  let ens: ENSRegistry
+  let registrar: FDSRegistrar
+
+  const record = {
+    node: 'placeholder',
+    owner: '0xC47c055b3EBfA044851Ac87B8240f77DF90fdcB8',
+    description: 'NFT Sample Dapp',
+    version: 1,
+    indexType: 1,
+    dataFormat: keccak256FromUtf8Bytes('beeson'),
+    blobRef: '0xfd79d5e0ebd8407e422f53ce1d7c4c41ebf403be55143900f8d1490560294780',
+    timestamp: new Date().getTime(),
+  }
+  before(async () => {
+    const [, controllerAccount] = await ethers.getSigners()
+    const ENSRegistry = await ethers.getContractFactory('ENSRegistry')
+    ens = await ENSRegistry.deploy()
+    await ens.deployed()
+
+    const FDSRegistrar = await ethers.getContractFactory('FDSRegistrar')
+    registrar = await FDSRegistrar.deploy(ens.address)
+    await registrar.addController(controllerAccount.address)
+    await ens.setSubnodeOwner(ZERO_HASH, keccak256FromUtf8Bytes('fds'), registrar.address)
+
+    const DappRegistry = await ethers.getContractFactory('DappRegistry')
+    // set ENS and FDS addresses, and configure subdomain (eg dappregistry.fds)
+    dappRegistry = await DappRegistry.deploy(ens.address, registrar.address, keccak256FromUtf8Bytes('dappregistry'))
+    await dappRegistry.deployed()
+    // set subnode owner and add to controller
+    await registrar.setSubnodeOwner(keccak256FromUtf8Bytes('dappregistry'), dappRegistry.address)
```

Comment:
I got error:
`Property 'setSubnodeOwner' does not exist on type 'FDSRegistrar'.ts(2339)`